NRLMMD-GEOIPS/geoips
# algorithm_interpolator_colormapper interpolation bug fix

# Reviewer Checklist


# Related Issues
No issues related to this one. Was found when developing the GLM reader (see #35)

# Testing Instructions
You can run full_test.sh to make sure everything is still working. As a side note, no products and/or product_defaults actually implement algorithm_interpolator_colormapper family. This was found because the GLM product used that family. This change fixed the bug when implementing that product.

# Summary
This was found during the development for the GLM reader. Apparently, at line 1033 of
single_source.py procflow, when trying to interpolate data under the
algorithm_interpolator_colormapper family, we call 'interp_plugin' with the same
input and output xarray. This makes no sense, and needs to be fixed. It wasn't changing
the data in this format. To fix it, we leave input_xarray as 'alg_xarray', and set
output_xarray to 'None'. This is handled properly in the interpolator, and will modify
coordinate arrays to fit the shape of the interpolated data. I've checked config_based,
and this doesn't occur for that procflow.

    modified: geoips/geoips/plugins/modules/procflows/single_source.py
